### PR TITLE
Removed scope guards where compiler rightgully emits warnings

### PR DESCRIPTION
**Pull request by cschlote**
I tested the package with the latest DMD build. It emits some warnings on scope guard and a nothrow attributes. 

Fixed up the code, so that is compiling properly.

### CODE REVIEW

File: source/semver.d [hunk 1]
```diff
@@ -132,7 +132,7 @@ struct SemVer
     /**
      * Return the canonical string format.
      */
-    string toString() const scope @safe
+    string toString() const @safe
```
Comment (Geod24) [inline source/semver.d:135]: Why was this removed ? Logically, it seems that `toString` should be `scope`. What prevents it from being `scope` ?

File: source/semver.d [hunk 2]
```diff
@@ -185,7 +185,7 @@ struct SemVer
         return result;
     }
 
-    private SemVer appendPrerelease0() scope @safe pure nothrow
+    private SemVer appendPrerelease0() @safe pure nothrow
```
Comment (Geod24) [inline source/semver.d:188]: `return scope` (I think) is the right way to do it here.

File: source/semver.d [hunk 3]
```diff
@@ -311,7 +311,7 @@ struct SemVer
                 return ids[i] < other.ids[i] ? -1 : 1;
         }
 
-        int compareSufix(const string[] suffix, const string[] anotherSuffix) @safe pure
+        int compareSufix(const scope string[] suffix, const string[] anotherSuffix) @safe pure
```
Comment (Geod24) [inline source/semver.d:314]: Feels like `anotherSuffix` could be `scope` too ?
Also prefer `scope const` over `const scope`, as `const` is a type qualifier and `scope` is a storage class.

File: source/semver.d [hunk 4]
```diff
@@ -372,11 +372,8 @@ struct SemVer
     /**
      * Compare two $(B different) versions and return the parte they differ on.
      */
-    VersionPart differAt(ref const SemVer other) const scope @safe pure nothrow
-    in
-    {
-        assert(this != other);
-    }
+    VersionPart differAt(ref const SemVer other) const scope @safe pure
+    in (this != other)
```
Comment (Geod24) [inline source/semver.d:376]: Unrelated change

File: source/semver.d [hunk 5]
```diff
@@ -395,7 +392,7 @@ struct SemVer
     }
 
     /// ditto
-    VersionPart differAt(const SemVer other) const scope @safe pure nothrow
+    VersionPart differAt(const SemVer other) const scope @safe pure
```
Comment (Geod24) [inline source/semver.d:395]: Unrelated ?

**Review by Geod24 — CHANGES_REQUESTED** [review]
Could use some explanation about the change, as they are not obvious to me.
Also, those are scope qualifiers, not scope guards.

Those are scope guards:
```D
scope (exit) {}
scope (failure) {}
scope (success) {}
```

File: source/semver.d (hunk 1 above)
Comment (cschlote) [reply source/semver.d:135]: You return a string created inside the function scope. The next dmd compiler complains about it, when scope is set. But returning a string is intended here. And the string is stored in the GC, so a 'scope' attribute isn't needed here.

Or the next dmd is wrong here...

File: source/semver.d (hunk 2 above)
Comment (cschlote) [reply source/semver.d:188]: So either you, or the next DMD is wrong.

File: source/semver.d (hunk 3 above)
Comment (cschlote) [reply source/semver.d:314]: Yes, reasonable.

File: source/semver.d (hunk 4 above)
Comment (cschlote) [reply source/semver.d:376]: Yes

File: source/semver.d (hunk 5 above)
Comment (cschlote) [reply source/semver.d:395]: Throwing a assert is not nothrow. 
Surely unreleated. I can splitt the PR into 3 separate.

You can check the latest DMD beta build, and think about, if DMD is wrong or the current code without the PR.

May intention is to get rid of the warnings.
